sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinLayoutController.java

```diff
@@ -324,10 +347,33 @@ public int getCurrentVolume() {
 
   @Override
   public void setFullscreen(boolean isFullscreen) {
+    if (_player != null && _player.hasVRContent()) {
+      if (_player.getVRMode() == VrMode.STEREO && !isFullscreen) {
+        _player.setVRMode(VrMode.MONO);
+      }
+
+      //Store screen orientation
+      storeScreenOrientation(isFullscreen);
+    }
+
     _layout.setFullscreen(isFullscreen);
     sendNotification(FULLSCREEN_CHANGED_NOTIFICATION_NAME, isFullscreen);
   }
 
+  private void storeScreenOrientation(boolean isFullScreen) {
+    boolean changed = isFullscreen() != isFullScreen;
+    Context context = getLayout().getContext();
+
+    if (isFullScreen) {
+      if (context instanceof Activity){
```

Comment:
I'm not sure if the `context` will always be an `Activity`, there could be cases in which it isn't, e..g. if the user sets the SkinLayout inside of `Fragment`.

Reply:
I believe that a fragment could be attached only within an activity.